cql3/expr/restrictions.cc

```diff
@@ -271,5 +271,24 @@ binary_operator validate_and_prepare_new_restriction(const binary_operator& rest
     return prepared_binop;
 }
 
+void validate_single_collection_column_relation(const expression& col_expr, oper_t oper, schema_ptr schema) {
```

Comment:
I think the name of this function is misleading. It doesn't "validate a collection column", it looks at any expression and looks for one specific problem it can have. I would make the name more generic, like validate_expression, with a comment explaining what it validates today (maybe next week we'll want it to validate more). 
I'm not even sure that it needs to be a separate function at all: It is only called once, and we can just insert that code in that one place. Making it a separate function in a separate source file just makes it harder to track, in my opinion.

Reply:
I used the function just to encapsulate validation logic and to make code of `prepare_binary_operator` more understandable. And put that function near other validation functions.
I could insert this code in one place if you think it's better for Scylla project style.